src/main/java/jfdi/parser/Constants.java

```diff
@@ -0,0 +1,98 @@
+package jfdi.parser;
+
+import java.time.ZoneId;
+
+public class Constants {
+
+    // ===============================
+    // Regex Constants
+    // ===============================
+
+    // General Regex Strings
+    public static final String REGEX_WHITESPACE = "\\s";
+
+    // Task type specific Regex Strings
+    public static final String REGEX_ADD = "^(add)";
+    public static final String REGEX_LIST = "^(list)";
+    public static final String REGEX_RENAME = "^(rename)";
+    public static final String REGEX_RESCHEDULE = "^(reschedule)";
+    public static final String REGEX_DELETE = "^(delete)";
+
+    // Task related Regex Strings
+    // Tags are always appended to the end of any input, hence the boundary
+    // condition $
+    public static final String REGEX_TAGS = "\\+\\w+$";
+    // A taskID is simply an integer
+    public static final String REGEX_TASKID = "\\d+";
+
+    // Date and Time related Regex Strings
+    public static final String REGEX_DAYS_NUMERIC = "(0?[1-9]|[12][\\d]|3[01])(st|th|nd|rd)?";
+    public static final String REGEX_DAYS_TEXTUAL = "((mon|MON)(day|DAY)?|"
```

Comment:
You can use case-insensitive regex to get rid of all this `(abc|ABC)` things :wink: 

You can use `java.util.regex.Pattern.compile()` with a `Pattern.CASE_INSENSITIVE` second argument to compile a case-insensitive pattern. 

Or, you can just use `(?i:(something))` to make the matching of `(something)` case-insensitive. 

For example, this regex can be changed to:
```
(?i:sun(day)?|mon(day)?|tue(sday)?|wed(nesday)?|thu(rsday)?|fri(day)?|sat(urday)?)
```